### PR TITLE
Dispatch events for Sorting / Sorted

### PR DESCRIPTION
**Pull request by lhilton**
I need my application to know when the sorting is either about to run or has run. This PR dispatches events both before and after the main sorting work in `SortableTrait::setNewOrder()`.

### CODE REVIEW

File: src/SortableTrait.php [hunk 1]
```diff
@@ -47,6 +49,8 @@ public static function setNewOrder($ids, int $startOrder = 1, string $primaryKey
         }
 
         $model = new static();
+        
+        Sorting::dispatch($model);
```
Comment (freekmurze) [inline src/SortableTrait.php:53]: Use the `event()` function to dispatch the new event.

File: tests/SortableTest.php [hunk 2]
```diff
@@ -371,4 +374,30 @@ public function it_can_tell_if_element_is_last_in_order()
...
+    /** @test */
+    public function it_dispatches_event_before_sort()
+    {
+        Event::fake();
+
+        $newOrder = Collection::make(Dummy::all()->pluck('id'))->shuffle()->toArray();
+
+        Dummy::setNewOrder($newOrder);
+
+        // Assert an event was dispatched twice...
```
Comment (freekmurze) [inline tests/SortableTest.php:387]: This comment does not seem to be correct

File: tests/SortableTest.php [hunk 3]
```diff
@@ -371,4 +374,30 @@ public function it_can_tell_if_element_is_last_in_order()
...
+    /** @test */
+    public function it_dispatches_event_after_sort()
+    {
+        Event::fake();
+
+        $newOrder = Collection::make(Dummy::all()->pluck('id'))->shuffle()->toArray();
+
+        Dummy::setNewOrder($newOrder);
+
+        // Assert an event was dispatched twice...
```
Comment (freekmurze) [inline tests/SortableTest.php:400]: This comment does not seem to be correct

**Review by freekmurze — CHANGES_REQUESTED** [review]
Thanks! Let's improve it a bit!

File: src/Events/Sorted.php [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    use Dispatchable, InteractsWithSockets, SerializesModels;
+
+    public Model $model;
+
+    /**
+     * Create a new event instance.
+     *
+     * @return void
+     */
+    public function __construct(Model $model)
```
[the remark this answers is not in the document]
Comment (lhilton) [reply src/Events/Sorted.php]: ✅

File: src/Events/Sorting.php [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+use Illuminate\Queue\SerializesModels;
+
+
+class Sorting
+{
+    use Dispatchable, InteractsWithSockets, SerializesModels;
+
+    public Model $model;
+
+    /**
```
[the remark this answers is not in the document]
Comment (lhilton) [reply src/Events/Sorting.php]: ✅

File: src/SortableTrait.php (hunk 1 above)
Comment (lhilton) [reply src/SortableTrait.php]: ✅

File: tests/SortableTest.php (hunk 2 above)
Comment (lhilton) [reply tests/SortableTest.php]: ✅

File: tests/SortableTest.php (hunk 3 above)
Comment (lhilton) [reply tests/SortableTest.php]: ✅

**Comment by lhilton** [comment]
Thank you @freekmurze for the feedback. I have updated the contribution per your guidelines, it is ready for re-review.

File: src/Events/Sorting.php [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+use Illuminate\Broadcasting\InteractsWithSockets;
+use Illuminate\Broadcasting\PresenceChannel;
+use Illuminate\Broadcasting\PrivateChannel;
+use Illuminate\Contracts\Broadcasting\ShouldBroadcast;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Foundation\Events\Dispatchable;
+use Illuminate\Queue\SerializesModels;
+
+
+class Sorting
```
Comment (freekmurze) [inline src/Events/Sorting.php:14]: Rename this to `SortingModel`

**Review by freekmurze — CHANGES_REQUESTED** [review]
One more nitpick and then I think it's ready to go!

File: src/Events/Sorted.php [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+use Illuminate\Broadcasting\InteractsWithSockets;
+use Illuminate\Broadcasting\PresenceChannel;
+use Illuminate\Broadcasting\PrivateChannel;
+use Illuminate\Contracts\Broadcasting\ShouldBroadcast;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Foundation\Events\Dispatchable;
+use Illuminate\Queue\SerializesModels;
+
+
+class Sorted
```
[the remark this answers is not in the document]
Comment (lhilton) [reply src/Events/Sorted.php:15]: ✅

File: src/Events/Sorting.php (hunk 6 above)
Comment (lhilton) [reply src/Events/Sorting.php:14]: ✅

**Comment by lhilton** [comment]
@freekmurze should be ready for re-review

File: src/Events/ModelSorted.php [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+use Illuminate\Broadcasting\PresenceChannel;
+use Illuminate\Broadcasting\PrivateChannel;
+use Illuminate\Contracts\Broadcasting\ShouldBroadcast;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Foundation\Events\Dispatchable;
+use Illuminate\Queue\SerializesModels;
+
+class ModelSorted
+{
+    use Dispatchable, InteractsWithSockets, SerializesModels;
```
Comment (freekmurze) [inline src/Events/ModelSorted.php:16]: These traits are not needed. Could you remove them from both events?

File: tests/SortableTest.php [hunk 9]
```diff
@@ -371,4 +374,30 @@ public function it_can_tell_if_element_is_last_in_order()
...
+    /** @test */
+    public function it_dispatches_event_before_sort()
+    {
+        Event::fake();
+
+        $newOrder = Collection::make(Dummy::all()->pluck('id'))->shuffle()->toArray();
+
+        Dummy::setNewOrder($newOrder);
+
+        // Assert the event was dispatched...
```
Comment (freekmurze) [inline tests/SortableTest.php:387]: This comment has no use, could you remove it?

**Review by freekmurze — CHANGES_REQUESTED** [review]
A few nitpicks!

**Pull request closed without merging**